sys/kern/time.c

```diff
@@ -47,4 +49,15 @@ time_t tm2sec(tm_t *t) {
   return (time_t)(t->tm_year - 70) * year_scale_s +
          (t->tm_mday - 1) * day_scale_s + t->tm_hour * hour_scale_s +
          t->tm_min * min_scale_s + t->tm_sec + res;
-}
\ No newline at end of file
+}
+
+static void mdelay_timeout(__unused void *arg) {
+  /* Nothing to do here. */
+}
+
+void mdelay(useconds_t ms) {
+  callout_t handle;
+  bzero(&handle, sizeof(callout_t));
+  callout_setup_relative(&handle, ms, mdelay_timeout, NULL);
+  callout_drain(&handle);
+}
```

Comment:
It's not clear for me. Argument for `mdelay` is ms (millisecond?) but `useconds_t` looks like microseconds. On the other hand second argument of `callout_setup_relative` is a `systime_t`. Could you explain?

Reply:
I'm having similar thoughts on this one. I'm be waiting for an explanation. Other than that, the logic seems fine, but I can't accept it unless you clarify what's going on here.